ADBench/plot_graphs.py

```diff
@@ -56,34 +58,54 @@
     "Zygote": ("y", "v", "Julia, Zygote")
 }
 
+# File with input data if specified
+if use_file:
+    arg_idx = sys.argv.index("--use-file") + 1
+    input_file_path = sys.argv[arg_idx]
+
 # Folders
 adbench_dir = os.path.dirname(os.path.realpath(__file__))
 ad_root_dir = os.path.dirname(adbench_dir)
-in_dir = os.path.join(ad_root_dir, "tmp")
+in_dir = os.path.join(ad_root_dir, "tmp") if not use_file else os.path.dirname(input_file_path)
 out_dir = os.path.join(in_dir, "graphs")
 static_out_dir_rel = "static"
 plotly_out_dir_rel = "plotly"
 static_out_dir = os.path.join(out_dir, static_out_dir_rel)
 plotly_out_dir = os.path.join(out_dir, plotly_out_dir_rel)
 
 # Scan folder for all files, and determine which graphs to create
-all_files = [path for path in utils._scandir_rec(in_dir) if TIMES_SUBSTRING in path[-1]]
-all_graphs = [path.split("/") for path in set(["/".join(path[:-2]) for path in all_files])]
-function_types = ["objective ÷ Manual", "objective", "jacobian", "jacobian ÷ objective"]
-all_graphs = [(path, function_type) for function_type in function_types for path in all_graphs]
-all_graph_dict = {}
+if not use_file:
+    all_files = [path for path in utils._scandir_rec(in_dir) if TIMES_SUBSTRING in path[-1]]
+    all_graphs = [path.split("/") for path in set(["/".join(path[:-2]) for path in all_files])]
+    function_types = ["objective ÷ Manual", "objective", "jacobian", "jacobian ÷ objective"]
+    all_graphs = [(path, function_type) for function_type in function_types for path in all_graphs]
```

Comment:
These variables are needed only in `main` under the `else` branch of an `if use_file` (`all_files` is used indirectly in the call of `get_sorted_vals_by_tool`) so these bindings should be moved there (and `all_files` passed as an argument to `get_sorted_vals_by_tool`).